src/Meshes/abstractMesh.ts

```diff
@@ -1624,13 +1624,27 @@ export class AbstractMesh extends TransformNode implements IDisposable, ICullabl
             // Shadow generators
             var generator = light.getShadowGenerator();
             if (generator) {
-                var shadowMap = generator.getShadowMap();
-
-                if (shadowMap && shadowMap.renderList) {
-                    meshIndex = shadowMap.renderList.indexOf(this);
+                if (generator.useCSM) {

```

Comment:
same here, I wonder if the array in none CSM could contain only one map to simplify those blocks

Reply:
erm, if I try to refactor this, one is a RTT, the other is an array. The compiler throws errors and the resulting code is longer than this. Gonna leave it for now.

Reply:
could extract the inside to a function ?

Reply:
ok, added a ```private removeMesh``` function and factored a little.